infra/enrolment-records/repository/src/main/java/com/simprints/infra/enrolment/records/repository/EnrolmentRecordRepositoryImpl.kt

```diff
@@ -137,19 +137,24 @@ internal class EnrolmentRecordRepositoryImpl @Inject constructor(
         onCandidateLoaded = onCandidateLoaded,
     )
 
-    private fun fromIdentityDataSource(dataSource: BiometricDataSource) = when (dataSource) {
-        is BiometricDataSource.Simprints -> localDataSource
+    private suspend fun fromIdentityDataSource(dataSource: BiometricDataSource) = when (dataSource) {
+        is BiometricDataSource.Simprints -> selectEnrolmentRecordLocalDataSource()
         is BiometricDataSource.CommCare -> commCareDataSource
     }
 
-    override suspend fun load(query: SubjectQuery): List<Subject> = localDataSource.load(query)
+    override suspend fun load(query: SubjectQuery): List<Subject> = selectEnrolmentRecordLocalDataSource().load(query)
 
-    override suspend fun delete(queries: List<SubjectQuery>) = localDataSource.delete(queries)
+    override suspend fun delete(queries: List<SubjectQuery>) = selectEnrolmentRecordLocalDataSource().delete(queries)
 
-    override suspend fun deleteAll() = localDataSource.deleteAll()
+    override suspend fun deleteAll() = selectEnrolmentRecordLocalDataSource().deleteAll()
 
     override suspend fun performActions(
         actions: List<SubjectAction>,
         project: Project,
-    ) = localDataSource.performActions(actions, project)
+    ) {
+        actions.filterIsInstance<SubjectAction.Creation>().forEach {
```

Comment:
This is problematic - what happens with the updates and deletes? They will modify records in the old DB but we may have already migrated those records. I understand that this is not expected as downsync is blocked and we don't currently support local updates or deletes. But at the very least I'd put an error log here if that's attempted so we know something got wrong.